tasks/new_e2e_tests.py

```diff
@@ -95,3 +102,91 @@ def run(ctx, profile="", tags=[], targets=[], configparams=[], verbose=True, cac
     if some_test_failed:
         # Exit if any of the modules failed
         raise Exit(code=1)
+
+
+@task
+def clean(_):
+    """
+    Clean any environment created with invoke tasks or e2e tests
+    """
+    if not _is_local_state(_get_pulumi_about()):
+        print("Cleanup supported for local state only, run `pulumi login --local` to switch to local state")
+        return
+
+    print("🧹 Clean up lock files")
+    lock_dir = os.path.join(Path.home(), ".pulumi", "locks")
+
+    for entry in os.listdir(Path(lock_dir)):
+        subdir = os.path.join(lock_dir, entry)
+        for filename in os.listdir(Path(subdir)):
+            file_path = os.path.join(subdir, filename)
+            if os.path.isfile(file_path) and filename.endswith(".json"):
+                os.remove(file_path)
+                print(f"🗑️ Deleted lock: {file_path}")
+
+    print("🧹 Clean up stacks")
+    stacks = _get_existing_stacks()
+
+    for stack in stacks:
+        print(f"🗑️ Cleaning up stack {stack}")
+        _destroy_stack(stack)
+
+
+def _get_existing_stacks() -> List[str]:
+    # ensure we deal with local stacks
+    output = subprocess.check_output(["pulumi", "stack", "ls", "--all"], cwd=tempfile.gettempdir())
+    output = output.decode("utf-8")
+    lines = output.splitlines()
+    lines = lines[1:]  # skip headers
+    e2e_stacks: List[str] = []
+    stack_name_prefix = _get_stack_name_prefix()
+    for line in lines:
+        stack_name = line.split(" ")[0]
+        # skip stacks created out of e2e tests
+        if stack_name.startswith(stack_name_prefix):
+            e2e_stacks.append(stack_name)
+    return e2e_stacks
+
+
+def _get_stack_name_prefix() -> str:
```

Comment:
I think you can remove this function as it is specific to the invoke tasks in test-infration-definition.

Reply:
Aren't stacks created in tests prefixed by the user name too ? Removing it